### ISSUE TITLE
Group or Tag Dependency 

### ISSUE THREAD

**Issue by alexnoyes** [checkly/public-roadmap#178]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Currently, if an API backend is down and you are monitoring x number of endpoints within a given group, all will fall to erroneous status and generate unnecessary alerting and management overhead. 
When configuring a group (or tag), there should be an option to add a monitor parent. This overarching API will be called the same as all other endpoints; on a schedule etc, but would be recommended to run on a quicker schedule. If this fails, it will pause all dependant monitors that have been associated with it. This could even become a webhook endpoint where you can point your application to notify checkly when something is back up and working to avoid the parent API from calling the service.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by tnolet** [comment]
@alexnoyes do I understand that the "avalanche" of alerts you receive is the main pain point here? I'm asking this because this is something we are aware of, and actually experience ourselves too. We monitor our API's of course with Checkly, and when we have an outage on our API, we also get quite a lot of messages.

**Comment by alexnoyes** [comment]
@tnolet that's right. 

In a simple example, if checkly is monitoring 5 API endpoints.

4 of these could be the services which run the application.
1 of these could be the application health check endpoint.

If the application health check endpoint is returning unhealthy, then you would expect one or more (or all) of the API service endpoints to be causing the unhealthy status.

In this case, it would be beneficial to have an optional dependency section for the hierarchy for each of the endpoints in a given group. 

So you have the option to configure API endpoint 1 to only check on an interval (as normal) if the status of API endpoint 5 is healthy. You can then build dependencies upon dependencies which further strengthens the alerting proposition as you can see what is the issue and what is 'paused' monitoring because of it.

**Comment by tnolet** [comment]
@alexnoyes thanks for explaining this. Another question. Do you use the Checkly groups feature? If you group checks for similar/the same endpoints, an option would be to do exclusive alerting for groups. i.e. the group goes into an alert state and effectively stops sending alerts for other checks that are in that group.

**Comment by alexnoyes** [comment]
@tnolet Yes this should be based around the group feature. I use Chekly groups, they're great. They provide a good amount of flexibility but they can be enhanced. The exclusivity would need to extend into an hierarchy, i.e nested groups. these nested groups could be dependent on a monitor from the parent group. This would then create a cascading effect.

Let me use an example of a school...

You might be monitoring the status of the school (1x Monitor):

- How responsive is it, how many students are there in total, how many classrooms are open in total, is it open, planned closure, true or false

You also segment the school and monitor floors (3x Monitors):

- How responsive is each of the 3 floors, how many students are on each floor, how many classrooms are open on each floor, planned closure, true or false

You also then monitor the classrooms themselves (9x Monitors)

- How responsive is each classroom, how many students in each of the 9 rooms, and issues in each classroom, planned closure, true or false

Now, if there was to be a fire on floor 3 and it was closed with as an unplanned closure, there would be 10x monitors 'affected' and alerting. 1x from the floor monitor and 9x from the classrooms within the floor. All other floors are unaffected. 

If the school was to close because the fire spread to all 3x floors, then we would expect 30 monitors to be 'affected' and alerting.

Using dependencies, all floor groups would be dependent on the overall school monitor and all rooms would be dependent on the floor groups. Therefore if there was a school closure, one alert would be generated and all child objects would be paused because of this.

Kind regards,
Alex

**Comment by tnolet** [comment]
@alexnoyes thanks. Reminds me of my Nagios days! 😅  I guess our mission is to make this still easy to reason about.

**Comment by modern-sapien** [comment]
@tnolet since you brought this up I am also ➕ 1️⃣

**Comment by randak** [comment]
+100 to this. Most of our checks depend on our login system. If our login system fails, we do not want to alert every single on-call engineer in the company for something they have no control over.

**Comment by tnolet** [comment]
@randak thanks for responding to this. I think we need to look at this, but a bit different than @alexnoyes explained it. I think we are discussing the abstract concept of a "service" that other "services" depend on the function. This is indeed a hierarchical relationship. If the auth service is down, no need to send out x amount of alerts for the website login feature. @umutuzgur FYI
